### DISCUSSION TITLE
Can't run flutter apps with 3rd-party plugins

### DISCUSSION THREAD

**Discussion by richprince23** [codemagic-ci-cd/codemagic-docs#1274] [Q&A] [answered]
Good day 
I am new to codemagic and i have been trying to build my flutter apps for ios

if i build an app without any dependencies in the pubspec.yaml, app runs correctly on the target devide.
But if there are dependencies, they open a white blank screen and quits.

Also, if i ssh/vnc into the build machine, i am able to run the app in the ios simulator, but i can't run the built app on target device
Been searching everywhere in the docs but 🤦‍♂️

Help please

**Comment by K-Codemagic** [comment]
We listed down a few known issues here - https://blog.codemagic.io/testfligh-crash/. Also I would suggest to check the device logs if the build is successful.

**Comment by richprince23** [comment, marked as answer]
My bad, i was building a debug version. I changed to release build mode and it runs perfectly.
Wasn't dependecncy-related
